EasyCommands/BlockHandlers/BlockHandlers.cs

```diff
@@ -71,9 +76,34 @@ public static class BlockHandlerRegistry {
                 return blockHandlers[blockType];
             }
 
-            public static List<Object> GetBlocks(Block blockType, Func<IMyTerminalBlock, bool> selector) => blockHandlers[blockType].GetBlocks(selector);
+            public static void invalidateCaches() {
+                PROGRAM.invalidBlockCache = true;
+                PROGRAM.invalidGroupCache = true;
+            }
+
+            public static List<IMyTerminalBlock> RetrieveBlocks() {
+                if (PROGRAM.invalidBlockCache) {
+                    PROGRAM.GridTerminalSystem.GetBlocks(PROGRAM.blockCache);
+                    PROGRAM.invalidBlockCache = false;
+                }
+                return PROGRAM.blockCache;
+            }
+
+            public static List<IMyBlockGroup> RetrieveGroups() {
+                if (PROGRAM.invalidGroupCache) {
+                    PROGRAM.GridTerminalSystem.GetBlockGroups(PROGRAM.groupCache);
+                    PROGRAM.invalidGroupCache = false;
+                }
+                return PROGRAM.groupCache;
+            }
 
```

Comment:
nit: I believe we can inline these into GetBLocks and GetBlocksInGroup, respectively